src/corehost/cli/fxr_resolver.cpp

```diff
@@ -53,6 +53,24 @@ namespace
     }
 }
 
+bool fxr_resolver::is_runtime_embedded()
+{
+    // Check for an export of DotNetRuntime pointing to a string of 'DotNetRuntime'
+    pal::dll_t exe;
+    if (!pal::get_current_executable(&exe))
+        return false;
+
+    const char *runtime_marker = u8"DotNetRuntime";
+    const char **marker_maybe = reinterpret_cast<const char**>(pal::get_symbol(exe, runtime_marker));
```

Comment:
If a user strips all symbols from a single-file app, will it cause problem for this detection logic? 
Should we instead search by byte-pattern, similar to the [apphost-dll marker](https://github.com/dotnet/core-setup/blob/master/src/corehost/corehost.cpp#L40-L42) and [bundle header](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/apphost/bundle/marker.cpp#L18-L28) ?

Reply:
Searching for the byte pattern is probably too expensive to do at runtime.

Reply:
After I run `strip --strip-all` on the test host, this logic can still get the export. Is that what you mean?

Reply:
Yes that's what I was asking. I guess, since it is an exported symbol, it is still expected to exist. right?

Reply:
We should not have to worry about this export being stripped out which Elinor confirmed.

Reply:
Thanks for confirming this.